### PR TITLE
FIX-#6459: support fastparquet>=2023.1.0

### LINKED ISSUE #6459: Unpin `fastparquet`
Motivation: https://github.com/modin-project/modin/pull/6442#issuecomment-1666379460

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #6459 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: environment-dev.yml [hunk 1]
```diff
@@ -37,7 +37,7 @@ dependencies:
   # pymssql==2.2.8 broken: https://github.com/modin-project/modin/issues/6429
   - pymssql>=2.1.5,!=2.2.8
   - psycopg2>=2.8.6
-  - fastparquet>=0.6.3,<2023.1.0
+  - fastparquet>=0.6.3
```
Comment (vnlitvinov) [inline environment-dev.yml:40]: I wonder if we should really bump the _lower_ bound instead, to something like `>2023.1.0`?.. as, if I understand correctly, earlier versions of `fastparquet` aren't properly compatible with pandas 2.0.

Comment (anmyachev) [reply environment-dev.yml:40]: On the one hand, yes, on the other hand, pandas still supports fairly old versions. I guess there are rare compatibility issues, so they don't raise the minimum version. I think we should do the same.

Comment (vnlitvinov) [reply environment-dev.yml:40]: Yeah, we should probably just mirror what pandas requires on `fastparquet`...

**Review by vnlitvinov — APPROVED** [review]
LGTM!

**Pull request merged**